test/factories/host_related.rb

```diff
@@ -90,6 +90,7 @@ def set_nic_attributes(host, attributes, evaluator)
   factory :nic_primary_and_provision, :parent => :nic_managed, :class => Nic::Managed do
     primary true
     provision true
+    domain
```

Comment:
Not specifying the domain on the interface made `FactoryGirl.create(:host, :managed)` fail at times. For example it'd try to create a host 'hostXXX' with domain 'domainYYY' and there was already an existing host with name 'hostXXX.domainYYY', so it didn't notice the name conflict until too late.